### PR TITLE
fix(ui): fully round button on loading state

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

Make button in `loading` state fully round

Before fix

Screenshot 2024-09-17 at 11 28 32

After fix

Screenshot 2024-09-17 at 11 28 45

This will fix the PUBLISH button in editor page, and the FOLLOW SPACE button
Also moved the `!px-0 w-[46px]` classes inside button component

### How to test

1. Go to a proposal creation page
2. The Publish button should be fully rounded when on loading state
3. Follow/Unfollow a space
4. The button should be fully rounded when on loading state

### CODE REVIEW

File: apps/ui/src/components/Ui/Button.vue [hunk 1]
```diff
@@ -19,7 +19,7 @@ withDefaults(
   <button
     :type="type"
     :disabled="disabled || loading"
-    :class="primary && 'primary'"
+    :class="{ primary: primary, 'w-[46px] !px-0': loading }"
```
Comment (ChaituVR) [inline apps/ui/src/components/Ui/Button.vue:22]: Could we remove `!` so any component can pass overrides if required

Also we are passing `px-3.5`
image

which is causing this jump
Sep-17-2024 17-11-12

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: apps/ui/src/components/Ui/Button.vue (hunk 1 above)
Comment (ChaituVR) [reply apps/ui/src/components/Ui/Button.vue:22]:
```suggestion
    :class="{ primary: primary, 'w-[46px] px-0': loading }"
```
This works

File: apps/ui/src/components/Ui/Button.vue [hunk 2]
```diff
@@ -19,8 +19,12 @@ withDefaults(
   <button
     :type="type"
     :disabled="disabled || loading"
-    :class="primary && 'primary'"
-    class="rounded-full leading-[100%] border button px-3.5 h-[46px] text-skin-link bg-skin-bg"
+    :class="{
+      primary: primary,
+      'w-[46px] px-0': loading,
+      'px-3.5': !loading || $attrs.class?.includes('w-full')
```
Comment (ChaituVR) [inline apps/ui/src/components/Ui/Button.vue:25]: But previously we were adding `px-3.5` even if there is no `w-full` 🤔 it may effect some button size?

Comment (ChaituVR) [inline apps/ui/src/components/Ui/Button.vue:25]:
```suggestion
      'px-3.5': !loading || ($attrs.class as 'string')?.includes('w-full')
```

**Review by ChaituVR — APPROVED** [review]
tAck

File: apps/ui/src/components/Ui/Button.vue (hunk 2 above)
Comment (wa0x6e) [reply apps/ui/src/components/Ui/Button.vue]: So finally, did this work ?

Comment (ChaituVR) [reply apps/ui/src/components/Ui/Button.vue]: it works

**Pull request merged**